packages/ppx/src/LensesPpx.re

```diff
@@ -164,77 +164,641 @@ let createSetLens = (~typeName, ~fields) => {
                                                                     loc,
+                                                                    }),
+                                                                  ppat_loc: loc,
+                                                                  ppat_attributes:
+                                                                    [],
+                                                                }),
+                                                              ),
+                                                          },
+                                                          pc_guard: None,
+                                                          pc_rhs: {
+                                                            pexp_loc: loc,
+                                                            pexp_attributes:
+                                                              [],
+                                                            pexp_desc:
+                                                              Pexp_match(
+                                                                {
+                                                                  pexp_desc:
+                                                                    Pexp_ident({
                                                                     txt:
                                                                     Lident(
                                                                     "value",
                                                                     ),
-                                                                  }),
-                                                              },
-                                                            ),
-                                                          ],
-                                                          Some({
+                                                                    loc,
+                                                                    }),
+                                                                  pexp_loc: loc,
+                                                                  pexp_attributes:
+                                                                    [],
+                                                                },
+                                                                [
+                                                                  {
+                                                                    pc_lhs: {
+                                                                    ppat_desc:
+                                                                    Ppat_construct(
+                                                                    {
+                                                                    txt:
```

Comment:
now with this additional code we reached the column size of text, I think it's a good idea to break this down into smaller functions